src/gluonts/nursery/autogluon_tabular.py

```diff
@@ -59,19 +61,25 @@ def get_prediction_dataframe(series):
     )
     convert_type = {x: "category" for x in df.columns.values[:4]}
     df = df.astype(convert_type)
-    return df
+
+    cache = [None] * (context_length + prediction_length) + list(series.values)
+    col = []
+    total = len(df)
+    for i in range(context_length):
+        col.append('lag' + str(i))
+        df['lag' + str(i)] = cache[i:i + total]
+    return df if train else df[-prediction_length:]
```

Comment:
I'm not sure the logic is very clear here: a few lines above you are setting up which columns are in the dataframe, and what values should be in them. Maybe you can do everything there?